fraud-detection-python/cloud-functions/geocode-addresses/main.py

```diff
@@ -0,0 +1,108 @@
+import base64
+import json
+import os
+import requests
+from urllib.parse import urlencode
+from google.cloud import bigquery
+
+dataset_name = 'invoice_parser_results'
+table_name = 'geocode_details'
+my_schema = [
+{
+   "name": "input_file_name",
+   "type": "STRING"
+ },{
+   "name": "entity_type",
+   "type": "STRING"
+ },{
+   "name": "entity_text",
+   "type": "STRING"
+ },{
+   "name": "place_id",
+   "type": "STRING"
+ },{
+   "name": "formatted_address",
+   "type": "STRING"
+ },{
+   "name": "lat",
+   "type": "STRING"
+ },{
+   "name": "lng",
+   "type": "STRING"
+ }
+]
+
+bq_client = bigquery.Client()
+ 
+def write_to_bq(dataset_name, table_name, geocode_response_dict, my_schema):
+ 
+  dataset_ref = bq_client.dataset(dataset_name)
+  table_ref = dataset_ref.table(table_name)
+  row_to_insert =[]
+  row_to_insert.append(geocode_response_dict)
+
+  json_data = json.dumps(row_to_insert, sort_keys=False)
+  #Convert to a JSON Object
+  json_object = json.loads(json_data)
+  print(json_object)
+  job_config = bigquery.LoadJobConfig(schema = my_schema)
+  job_config.source_format = bigquery.SourceFormat.NEWLINE_DELIMITED_JSON
+
+  job = bq_client.load_table_from_json(json_object, table_ref, job_config=job_config)
+  error = job.result()  # Waits for table load to complete.
+  print(error)
+  
+ 
+def process_address(event, context):
+  """Triggered from a message on a Cloud Pub/Sub topic.
+  Args:
+  event (dict): Event payload.
+  context (google.cloud.functions.Context): Metadata for the event.
+  """
+  pubsub_message = base64.b64decode(event['data']).decode('utf-8')
+  #print(type(pubsub_message))
+  #print(pubsub_message)
+  message_dict = json.loads(pubsub_message)
+  query_address = message_dict.get('entity_text')
+  geocode_dict = {} 
+  geocode_dict["input_file_name"] = message_dict.get('input_file_name')
+  geocode_dict["entity_type"] = message_dict.get('entity_type')
+  geocode_dict["entity_text"] = query_address
+  geocode_response_dict = extract_geocode_info(query_address)
+  geocode_dict.update(geocode_response_dict)
+  print(geocode_dict)
+
+  write_to_bq(dataset_name, table_name, geocode_dict, my_schema)
+  #print(geocode_response_dict)
```

Comment:
Nit: i would remove this since it's commented out

Reply:
Done